jest.conf/testUtils.js

```diff
@@ -29,3 +22,56 @@ export const testAfterResize = testFunction => {
   };
   animationFrameId = requestAnimationFrame(assertAfterResize);
 };
+
+export async function renderComponent(page, component, props) {
+  await page.evaluate(
+    ({ component, props }) => {
+      window.postMessage(
+        {
+          type: 'RENDER_COMPONENT',
+          component: component,
+          props: props,
+        },
+        '*'
+      );
+    },
+    { component, props }
+  );
+  await page.waitForSelector('#testing-playground');
+
+  const isComponentRendered = await page.evaluate(() => {
+    const testing_playground = document.querySelector('#testing-playground');
+    return testing_playground && testing_playground.children.length > 0;
+  });
+
+  if (!isComponentRendered) {
+    // eslint-disable-next-line no-console
+    console.error('Component did not render in the testing playground');
+  }
+}
+
+export async function takeSnapshot(page, name) {
+  if (process.env.TEST_TYPE == 'visual') {
+    await percySnapshot(page, name);
+  } else {
+    // eslint-disable-next-line no-console
+    console.log(`Skipping Percy snapshot: ${name}`);
+  }
+}
+
+export function describeUnit(name, fn) {
```

Comment:
If we consider the additional overhead of migrating all the tests, you are right. I had concerns about that too, bu since the idea was okayed in the meetings, I chose to go along with it. 
I'll shift this to use the [Visual] tags, but we'd have to see a lot of tests marked as skipped in the console output. One way to handle that would be to use a `custom reporter`. Or we can just use `verbose=false` with the test commands to reduce the output verbosity in that case. 
Let me know what you'd prefer!

Reply:
Oh, Im sorry, a thousand apologies if we okayed you in that direction, there was probably some misunderstanding on my side :').

Regarding the test outputs. For now I think there would be no problems in seeing the skipped tests, I was testing it and I see that Jest does report all the non-skipped test suites grouped at the end, and if there is an error it always reports them all at the end too, so they still have visibility. However if it becomes a problem I think we could consider the custom reporter option. Would love to know @bjester's opinion on that too.

Reply:
Oh no problem, I should have understood the requirements clearly. 
For now I'll update the PR with the required changes. We can discuss and work on this later if the need arises.